### PR TITLE
Bug 1181061 - Add support for calendar events with zero duration

### PR DESCRIPTION
**Pull request by AlekoNN**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4fa437a495f30caf2f5d593b089a02000e06993c

File: apps/calendar/js/views/modify_event.js [hunk 1]
```diff
@@ -397,6 +397,16 @@ ModifyEvent.prototype = {
       endTime
     );
 
+    // Yahoo uses the same start and end date & time for all day events.
+    // In order to distinguish all-day event from Yahoo calendar and event
+    // with 0 duration we set duration of the event starting and ending at
+    // the same date and time to 1 millisecond.
+    if (fields.endDate.getTime() == fields.startDate.getTime() && !allday) {
```
Comment (gaye) [inline apps/calendar/js/views/modify_event.js]: nit: use `===` and check for the simpler condition first ie

```js
if (!allday && fields.endDate.getTime() === fields.startDate.getTime()) {
  // ...
}
```

File: apps/calendar/js/views/modify_event.js [hunk 2]
```diff
@@ -588,7 +598,7 @@ ModifyEvent.prototype = {
         if (targetElement.id === 'start-date-locale' ||
             targetElement.id === 'start-time-locale') {
           this._setEndDateTimeWithCurrentDuration();
-        } else if (this._getEndDateTime() <= this._getStartDateTime()) {
+        } else if (this._getEndDateTime() < this._getStartDateTime()) {
```
Comment (gaye) [inline apps/calendar/js/views/modify_event.js]: Please update the comment above (on L595) to match the new logic

File: apps/calendar/js/views/single_day.js [hunk 3]
```diff
@@ -120,6 +120,13 @@ SingleDay.prototype = {
     var duration = relativeDuration(this.date, startDate, endDate);
     // we subtract border to keep a margin between consecutive events
     var hei = duration * this._hourHeight - this._borderWidth;
+
+    // It doesn't make sense to set height less than 1 px, so we set it to 1 px
+    // in order to show events correctly.
+    if (hei < 1) {
```
Comment (gaye) [inline apps/calendar/js/views/single_day.js]: nit: `hei = Math.min(1, hei);`

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e7a06e83735e7cc1f5ab0b8a374172364b23be89

**Comment by AlekoNN** [comment]
Thanks for review, new commit with the required changes has been uploaded.

**Pull request merged**